18_regexp_lab/01.py

```diff
@@ -1,12 +1,14 @@
-"""
-Write a program that reads data
-from property files.
-Each line in the file can either be:
-    An empty line
-    A comment line (Start with #)
-    A property line (of the form key = value)
-
-Write a program that takes a property file name and key
-as command line arguments and prints the requested value
-"""
-
+#commited by shlomit
+import sys
+import re
+if len(sys.argv) < 3: 
+    sys.exit("error - you didn't add any arguments!")
+
+regex= sys.argv[2] + '[ ]*=[ ]*(.*)'
+
+file = sys.argv[1]
+with open(file, "r") as f:
+    for line in f:
+        res = re.search(regex,line)

```

Comment:
Using external input as regexp is considered bad practice: The input may contain characters that have special meaning in regexp, such as `.`, `^`, `$`, etc. Better to read all values to a dictionary and search that.